cli/cmd/ui.go

```diff
@@ -55,222 +50,104 @@ var uiCmd = &cobra.Command{
+	dialer, err := createDialer(method, url, cfg)
 	if err != nil {
-		return fmt.Errorf("failed to create gzip reader: %v", err)
+		return err
 	}
-	defer gzipReader.Close()
-
-	// Step 3: Create a new tar reader
-	tarReader := tar.NewReader(gzipReader)
-
-	// Step 4: Extract files one by one
-	for {
-		header, err := tarReader.Next()
-		if err == io.EOF {
-			break // Reached the end of the tar archive
-		}
-		if err != nil {
-			return fmt.Errorf("failed to read tar header: %v", err)
-		}
-
-		// Step 5: Create the file or directory
-		targetPath := filepath.Join(dir, header.Name)
-		if header.FileInfo().IsDir() {
-			if err := os.MkdirAll(targetPath, 0755); err != nil {
-				return fmt.Errorf("failed to create directory: %v", err)
-			}
-			continue
-		}
 
-		file, err := os.Create(targetPath)
-		if err != nil {
-			return fmt.Errorf("failed to create file: %v", err)
-		}
+	fw, err := portforward.NewOnAddresses(dialer,
+		[]string{"localhost"},
+		[]string{defaultPort}, stopCh, readyCh, nil, os.Stderr)
 
-		// Step 6: Copy file contents from tar to the target file
-		if _, err := io.Copy(file, tarReader); err != nil {
-			file.Close()
-			return fmt.Errorf("failed to extract file: %v", err)
-		}
-
-		file.Close()
+	if err != nil {
+		return err
 	}
-
-	return os.Chmod(filepath.Join(dir, "odigos-ui"), 0755)
-}
-
-func getDownloadUrl(os string, arch string, version string) string {
-	return fmt.Sprintf(uiDownloadUrl, version, version, os, arch)
+	return fw.ForwardPorts()
 }
 
-type Release struct {
-	TagName string `json:"tag_name"`
-}
+func findOdigosUIPod(client *kube.Client, ctx context.Context, ns string) (*corev1.Pod, error) {
+	pods, err := client.CoreV1().Pods(ns).List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app.kubernetes.io/name=%s", resources.UIAppLabelValue),
+	})
 
-func GetLatestReleaseVersion() (string, error) {
-	url := "https://api.github.com/repos/keyval-dev/odigos/releases/latest"
-
-	timeout := time.Duration(3 * time.Second)
-	client := http.Client{
-		Timeout: timeout,
-	}
-	resp, err := client.Get(url)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("failed to fetch latest release: %s", resp.Status)
+	if len(pods.Items) != 1 {
+		return nil, fmt.Errorf("expected to get 1 pod got %d", len(pods.Items))
 	}
 
-	var release Release
-	err = json.NewDecoder(resp.Body).Decode(&release)
-	if err != nil {
-		return "", err
+	pod := &pods.Items[0]
+	if pod.Status.Phase != corev1.PodRunning {
+		return nil, fmt.Errorf("odigos-ui pod is not running")
 	}
 
-	ver, _ := strings.CutPrefix(release.TagName, "v")
-	return ver, nil
+	return &pods.Items[0], nil
 }
 
 func init() {
 	rootCmd.AddCommand(uiCmd)
-	uiCmd.Flags().String("address", "localhost", "address to listen on")
-	uiCmd.Flags().Int("port", defaultPort, "port to listen on")
```

Comment:
what if port 3000 is already taken on the host? don't we want the user to allow binding to other port?